examples/src/main/java/kafka/examples/ExactlyOnceMessageProcessor.java

```diff
@@ -170,6 +148,43 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
         latch.countDown();
     }
 
+    private int processMessages()
+        throws ProducerFencedException, FencedInstanceIdException {
+        try {
+            ConsumerRecords<Integer, String> records = consumer.poll(Duration.ofMillis(200));
+            if (records.count() > 0) {
+                // Begin a new transaction session.
+                producer.beginTransaction();
+                for (ConsumerRecord<Integer, String> record : records) {
+                    // Process the record and send to downstream.
+                    ProducerRecord<Integer, String> customizedRecord = transform(record);
+                    producer.send(customizedRecord);
+                }
+                Map<TopicPartition, OffsetAndMetadata> positions = new HashMap<>();
+                for (TopicPartition topicPartition : consumer.assignment()) {
+                    positions.put(topicPartition, new OffsetAndMetadata(consumer.position(topicPartition), null));
+                }
+                // Checkpoint the progress by sending offsets to group coordinator broker.
+                // Under group mode, we must apply consumer group metadata for proper fencing.
+                if (this.mode.equals("groupMode")) {
+                    producer.sendOffsetsToTransaction(positions, consumer.groupMetadata());
```

Comment:
Could we do "group mode" only in this example? The example doesn't really extend to multiple instances otherwise.

Reply:
Not sure I follow, we could manually compute the partitions to assign to for the standalone mode?